### DISCUSSION TITLE
Feature Request: HRM overlay over Power

### DISCUSSION THREAD

**Discussion by effectpears** [dvmarinoff/Auuki#202] [Ideas]
I would be happy if there was an option to display the pulse curve as a overay of the powerchart ​​in the "Power" field

**Comment by dvmarinoff** [comment]
Sure, I like the idea. I've been meaning to re-work the power graph for some time now. I want to include real time graphs for SmO2 and THb, will add Heart Rate overlay also when I get to that task.

**Comment by dvmarinoff** [comment]
I have some updates on this. My idea is to make the graph switchable. You'd be able to click on the graph and switch to the next one.

### First graph:
- will be the familiar power bar graph
- it shows power as it comes live from the Trainer without being attached to the running clock and draws new values even when the clock is paused

### Second graph:
-  will include Heart Rate, Power, SmO2, THb
- this one will be attached to the running clock and will draw new values only when the clock is running

Here is how the second one looks currently:
- Red is Heart Rate
- Yellow is Power
- Orange is THb
- Green is SmO2

<img width="612" alt="Screenshot 2025-02-11 at 18 42 08" src="https://github.com/user-attachments/assets/c1c6c8fc-74c1-404a-872a-ec09bbfd1b31" />

**Comment by Xiaorongma** [comment]
What is the hardware implementation?

**Reply by dvmarinoff** [reply]
Not following, can you elaborate?
